api/src/main/java/org/pmiops/workbench/institution/InstitutionServiceImpl.java

```diff
@@ -1,103 +1,113 @@
-import org.pmiops.workbench.db.model.DbInstitutionEmailAddress;
-import org.pmiops.workbench.db.model.DbInstitutionEmailDomain;
+import org.pmiops.workbench.db.model.DbVerifiedInstitutionalAffiliation;
 import org.pmiops.workbench.model.Institution;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Import;
 import org.springframework.stereotype.Service;
 
 @Service
+@Import({InstitutionMapperImpl.class})
 public class InstitutionServiceImpl implements InstitutionService {
 
   private final InstitutionDao institutionDao;
+  private final InstitutionMapper institutionMapper;
+  private final VerifiedInstitutionalAffiliation verifiedInstitutionalAffiliation;
 
   @Autowired
-  InstitutionServiceImpl(InstitutionDao institutionDao) {
+  InstitutionServiceImpl(
+      InstitutionDao institutionDao,
+      InstitutionMapper institutionMapper,
+      VerifiedInstitutionalAffiliation verifiedInstitutionalAffiliation) {
     this.institutionDao = institutionDao;
+    this.institutionMapper = institutionMapper;
+    this.verifiedInstitutionalAffiliation = verifiedInstitutionalAffiliation;
   }
 
   @Override
   public List<Institution> getInstitutions() {
     return StreamSupport.stream(institutionDao.findAll().spliterator(), false)
-        .map(this::toModel)
+        .map(institutionMapper::dbToModel)
         .collect(Collectors.toList());
   }
 
   @Override
   public Optional<Institution> getInstitution(final String shortName) {
-    return getDbInstitution(shortName).map(this::toModel);
+    return getDbInstitution(shortName).map(institutionMapper::dbToModel);
+  }
+
+  @Override
+  public Optional<DbInstitution> getDbInstitution(final String shortName) {
+    return institutionDao.findOneByShortName(shortName);
   }
 
   @Override
   public Institution createInstitution(final Institution institutionToCreate) {
-    return toModel(institutionDao.save(newDbObject(institutionToCreate)));
+    return institutionMapper.dbToModel(
+        institutionDao.save(institutionMapper.modelToDb(institutionToCreate)));
   }
 
   @Override
-  public boolean deleteInstitution(final String shortName) {
+  public DeletionResult deleteInstitution(final String shortName) {
     return getDbInstitution(shortName)
         .map(
             dbInst -> {
-              institutionDao.delete(dbInst);
-              return true;
+              if (verifiedInstitutionalAffiliation.findAllByInstitution(dbInst).isEmpty()) {
+                // no verified user affiliations: safe to delete
+                institutionDao.delete(dbInst);
+                return DeletionResult.SUCCESS;
+              } else {
+                return DeletionResult.HAS_VERIFIED_AFFILIATIONS;
+              }
             })
-        .orElse(false);
+        .orElse(DeletionResult.NOT_FOUND);
   }
 
   @Override
   public Optional<Institution> updateInstitution(
       final String shortName, final Institution institutionToUpdate) {
```

Comment:
This enables updating the inst's ShortName if necessary.

update("Broad", inst("TheBroadInst", ...)) would change Broad's shortName to TheBroadInst.  I can add a test to make that more clear.

Reply:
These were the principles driving this API:
* Don't expose your DB key
* An institute may need to modify its Short Name

Maybe these were the wrong priorities or this could have been implemented a different way.  I'm open to looking at this again as a followup.